### DISCUSSION TITLE
[Further] 关于web控制面板的讨论

### DISCUSSION THREAD

**Discussion by majoson-chen** [pymumu/smartdns#818] [Show and tell]
## 前言

前面已经在 issue 中提到过给 smartdns(以下简称sdns) 做一个通用 web 控制面板的想法. 以下详细说明一下个人看法.

## 概述

  sdns 在openwrt 中已经有一个 web 面板了, 我不清楚 luci 插件开发的流程, 但是我想 luci 应该是不需要开发者过多干涉 web 业务这一块的, 这意味着这个 web 面板只能在 luci 中使用(无法在其他 linux 发行版中使用), 而且这个 web 面板存在诸多问题.

  针对这样的问题, 我的提议是将 sdns 的核心与 web (配置接口)业务分离开, 核心只需要专注于域名的查询和解析就ok了, 其他的交由另外的程序来把控(譬如 clash 就是这样做的)

  这样做不仅可以保证核心业务的稳定和干净, 在处理性能不变的同时又使之变得更加易用, 可拓展性又上升了一个级别.

## 实现

  我将会使用 python 做API后台(因为我不太熟悉其他语言), 开发一套前后端分离的面板, 同时移除在 luci 中的设置功能, 只留下一个 smartdns 的业务开关, 随后所有业务都由这一个面板来承担, 这样就保证了在所有平台的可用性.

  如果你有兴趣, 我会补充更多的细节.

**Comment by majoson-chen** [comment]
..hi?

**Reply by emacsnt** [reply]
you may discuss this feature on issues, if you put the topic here in discussion, the author may not be notified.

**Reply by majoson-chen** [reply]
啊这..
